### PR TITLE
Feature/blured background on detend

### PR DESCRIPTION
**Pull request by sladan-hedvig**
- Added blured background as option for detent

## Checklist

- [ ] Dark/light mode verification
- [ ] Landscape verification
- [ ] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

### CODE REVIEW

**Comment by sladan-hedvig** [comment]
Just add .bluredBackground as option on journeys

File: Projects/hCoreUI/Sources/PresentationStyle+Detented.swift [hunk 1]
```diff
@@ -140,12 +150,34 @@ class DetentedTransitioningDelegate: NSObject, UIViewControllerTransitioningDele
...
     ) -> UIPresentationController? {
-        let key = ["_", "U", "I", "Sheet", "Presentation", "Controller"]
-        let sheetPresentationController = NSClassFromString(key.joined()) as! UIPresentationController.Type
-        let presentationController = sheetPresentationController.init(
-            presentedViewController: presented,
-            presenting: presenting
-        )
+
+        let presentationController: UIPresentationController = {
+            if #available(iOS 15.0, *), options.contains(.bluredBackground) {
```
Comment (julia-andersson) [inline Projects/hCoreUI/Sources/PresentationStyle+Detented.swift:155]: just a typo: should be .blurredBackground

File: Projects/hCoreUI/Sources/PresentationStyle+Detented.swift [hunk 2]
```diff
@@ -140,12 +150,34 @@ class DetentedTransitioningDelegate: NSObject, UIViewControllerTransitioningDele
...
+                return BlurredSheetPresenationController(
+                    presentedViewController: presented,
+                    presenting: presenting,
+                    useBlur: options.contains(.bluredBackground)
```
Comment (julia-andersson) [inline Projects/hCoreUI/Sources/PresentationStyle+Detented.swift:159]: same

File: Projects/hCoreUI/Sources/PresentationStyle+Detented.swift [hunk 3]
```diff
@@ -140,12 +150,34 @@ class DetentedTransitioningDelegate: NSObject, UIViewControllerTransitioningDele
...
+                let sheetPresentationController = NSClassFromString(key.joined()) as! UIPresentationController.Type
+                let presentationController = sheetPresentationController.init(
+                    presentedViewController: presented,
+                    presenting: presenting
+                )
+                return presentationController
+            }
+        }()
+
+        if #available(iOS 16.0, *), options.contains(.bluredBackground) {
```
Comment (julia-andersson) [inline Projects/hCoreUI/Sources/PresentationStyle+Detented.swift:172]: same

File: Projects/hCoreUI/Sources/PresentationStyle+Detented.swift [hunk 4]
```diff
@@ -206,6 +238,7 @@ class DetentedTransitioningDelegate: NSObject, UIViewControllerTransitioningDele
 extension PresentationOptions {
     // adds a grabber to DetentedModals
     public static let wantsGrabber = PresentationOptions()
+    public static let bluredBackground = PresentationOptions()
```
Comment (julia-andersson) [inline Projects/hCoreUI/Sources/PresentationStyle+Detented.swift:241]: blurredBackground

**Review by julia-andersson — CHANGES_REQUESTED** [review]

**Review by julia-andersson — APPROVED** [review]

**Pull request merged**